src/Storages/MergeTree/MergeTreeIndexAnnoy.cpp

```diff
@@ -0,0 +1,290 @@
+#ifdef ENABLE_ANNOY
+
+#include <Storages/MergeTree/MergeTreeIndexAnnoy.h>
+
+#include <Common/typeid_cast.h>
+#include <Core/Field.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+
+
+namespace DB
+{
+
+namespace Annoy
+{
+
+template<typename Dist>
+void AnnoyIndexSerialize<Dist>::serialize(WriteBuffer& ostr) const
+{
+    if (!Base::_built)
```

Comment:
If this is not copy-pasted from elsewhere  (in which case we could leave it as is), I would prefer to put the sanity check into an assert: `assert(Base::_build)`